src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -475,7 +475,10 @@ private BoundStatement BindGoto(GotoStatementSyntax node, DiagnosticBag diagnost
                     if (binder == null)
                     {
                         Error(diagnostics, ErrorCode.ERR_InvalidGotoCase, node);
-                        return new BoundBadStatement(node, ImmutableArray<BoundNode>.Empty, true);
+                        var childNodes = node.Expression != null ?
+                            ImmutableArray.Create<BoundNode>(BindExpression(node.Expression, diagnostics)) :
```

Comment:
>diagnostics [](start = 93, length = 11)

Consider passing new diagnostics bag to suppress any diagnostics from this binding